keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,973 @@
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+If a user sets a PodDisruptionBudget (PDB) for some low-priority pods and sets the `AllowDisruptionByPriorityGreaterThanOrEqual` 
+to `PriorityClass`, high-priority (less than the value of allowDisruptionByPriorityGreaterThanOrEqual) pods will not be able 
+to violate the PDB and preempt these pods during the scheduling process. This may result in high-priority (less than the value of allowDisruptionByPriorityGreaterThanOrEqual) pods being unable to schedule while low-priority pods continue to run normally. 
+
+Although the above situations may arise, due to the fact that PriorityClass is created and managed by 
+cluster administrators with no permission for application owners to perform actions, administrators 
+are able to uniformly configure according to the requirements. Additionally, implementation will 
+include the addition of additional logging or event descriptions to clearly inform the user of the 
+reason why preemption did not occur.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+In order to address the issue mentioned above, a new field `AllowDisruptionByPriorityGreaterThanOrEqual` will 
+be added to `PriorityClass`. Users will be able to set this field to indicate that the PodDisruptionBudget of 
+the pods associated with this priority class can only be violated during scheduler preemption by other pods 
+with a priority value greater than or equal to AllowDisruptionByPriorityGreaterThanOrEqual.
+The default value for this field will be `Value` + 1, which is consistent with current behavior.  
+
+```go
+type PriorityClass struct {
+  metav1.TypeMeta
+  metav1.ObjectMeta
+  Value int32
+  GlobalDefault bool
+  Description string
+  PreemptionPolicy *core.PreemptionPolicy
+
+  // AllowDisruptionByPriorityGreaterThanOrEqual indicates that the PodDisruptionBudget of the pods 
+  // associated with this priority class can only be violated during scheduler preemption by other pods 
+  // with a priority value greater than or equal to AllowDisruptionByPriorityGreaterThanOrEqual.
+  // Defaults to Value+1 if unset.
```

Comment:
We shouldn't default to it. Instead, there should be an interpretation for null.

This is because users might want to change the `Value` field without having to worry about `AllowDisruptionBy...` if they didn't initially change it.

Reply:
Yes, you are right, I overlooked the possibility that the user might modify the priority value.
Added the interpretation for null.